conans/client/conan_api.py

```diff
@@ -800,6 +801,7 @@ def package(self, path, build_folder, package_folder, source_folder=None, instal
         conanfile.folders.set_base_source(source_folder)
         conanfile.folders.set_base_package(package_folder)
         conanfile.folders.set_base_install(install_folder)
+        conanfile.folders.set_base_generators(conanfile_dir)
```

Comment:
Not a big deal, if ``conan package`` is removed in Conan 2.0, but this also relies on the conanfile location, not an extra ``--build-folder`` location.